src/dmd/parse.d

```diff
@@ -6743,6 +6751,12 @@ final class Parser(AST) : Lexer
         return parseAssignExp();
     }
 
+    private void check()
+    {
+        error("incorrect instantiation of `%s`", token.toChars());
```

Comment:
This error message is not really helpful, maybe smth like this is more appropriate:

Given `Foo foo(42);`
```
foo.d(1): Found `(` when expecting `;` or `=`, did you mean `Foo foo = 42`?
```

Reply:
Thank you for your feedback, I made the modifications!